Adds instructions to readme about JMXFetch and clarifies detailed dev docs (#20514)
DataDog/datadog-agent: docs/dev/checks/jmxfetch.md

integrations locally. In other words, you've made changes to how JMXFetch
works and you'd like to run JMXFetch via the Agent to validate your changes.

## Stable JMXFetch

> If your goal is just to run an JMX-based check and you don't need a custom
> version of JMXFetch, follow these instructions.

Reviewer: ```suggestion
If your goal is to run a JMX-based check and you don't need a custom
version of JMXFetch, follow the instructions below:
```

1. Download the `-jar-with-dependencies.jar` build of the latest version of JMXFetch from
   [`maven`](https://repo1.maven.org/maven2/com/datadoghq/jmxfetch/)

> 2. Copy it to `$GOPATH/src/github.com/DataDog/datadog-agent/dev/dist/jmx/jmxfetch.jar`.

Reviewer 2: Maybe Copy *and rename* to . . .

> 2. Copy it to `$GOPATH/src/github.com/DataDog/datadog-agent/dev/dist/jmx/jmxfetch.jar`.

Reviewer: ```suggestion
2. Copy the jar file and rename it to `$GOPATH/src/github.com/DataDog/datadog-agent/dev/dist/jmx/jmxfetch.jar`.
```

3. Run `inv agent.run`.
4. Validate that the JMXFetch section appears in `agent status`.

> If you have a JMX-based integration configured to run, it should automatically
> be run in your local JMXFetch instance.

Reviewer: ```suggestion
If you have a JMX-based integration configured to run, it automatically
runs in your local JMXFetch instance.
```

> 

Reviewer 3: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ 
```suggestion
```
